<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name="input" default="/camera/color/image_raw"/>
    <arg name="output" default="/lseg/pred"/>
    <arg name="weights" default="$(find lang_seg_ros)/checkpoints/demo_e200.ckpt"/>
    <arg name="drop" default="true"/>
    <arg name="labels"  
        default="gravel,road,bench,garbage bin,grass,building,sky,door,car,person,bush,fence,gate,tree,chair,table,robot,sign,sidewalk,other"/>


    <node pkg="image_transport" type="republish"  name="uncompress"
        args="compressed in:=$(arg input) raw out:=$(arg input)">
    </node>

    <node pkg="lang_seg_ros" type="lseg_infer.py" name="lang_seg_ros" output="screen">
        <param name="input" value="$(arg input)"/>
        <param name="output" value="$(arg output)"/>
        <param name="weights" value="$(arg weights)"/>
        <param name="drop" value="$(arg drop)"/>
        <param name="labels" value="$(arg labels)"/>
    </node>
</launch>
